modules/scripting/src/main/java/org/mule/module/scripting/transformer/ScriptTransformer.java

```diff
@@ -22,6 +24,20 @@
     private Scriptable script;
 
     @Override
+    public void initialise() throws InitialisationException
+    {
+        super.initialise();
+        LifecycleUtils.initialiseIfNeeded(script, muleContext);
+    }
+
+    @Override
+    public void dispose()
+    {
+        super.dispose();
+        LifecycleUtils.disposeIfNeeded(script);
```

Comment:
if you pass a logger, then it uses that logger to log an eventual exception. otherwise, it uses one of its own. It's just a cosmetic feature to have the proper category shown in the logs